pkg/peer/peer.go

```diff
@@ -246,3 +217,32 @@ func (p *Peer[ID]) ProcessSDPOffer(sdpOffer string) (*webrtc.SessionDescription,
 
 	return &answer, nil
 }
+
+// Read incoming RTCP packets
+// Before these packets are returned they are processed by interceptors. For things
+// like NACK this needs to be called.
+func (p *Peer[ID]) readRTCP(rtpSender *webrtc.RTPSender) {
+	for {
+		packets, _, err := rtpSender.ReadRTCP()
+		if err != nil {
+			if errors.Is(err, io.ErrClosedPipe) || errors.Is(err, io.EOF) {
+				p.logger.WithError(err).Warn("failed to read RTCP on track")
+				return
+			}
+		}
+
+		// We only want to inform others about PLIs and FIRs. We skip the rest of the packets for now.
+		toForward := []RTCPPacketType{}
+		for _, packet := range packets {
+			// TODO: Should we also handle NACKs?
+			switch packet.(type) {
+			case *rtcp.PictureLossIndication:
+				toForward = append(toForward, PictureLossIndicator)
+			case *rtcp.FullIntraRequest:
+				toForward = append(toForward, FullIntraRequest)
+			}
+		}
+
+		p.sink.Send(RTCPReceived{Packets: toForward, TrackID: rtpSender.Track().ID()})
+	}
+}
```

Comment:
I think I might have moved it just to keep the order of functions as "first public functions/interface, then private functions", i.e. `WriteRTCP()` is a public function (Go is case sensitive: functions, types, fields that start with capital are exported, whereas those written with a lowercase are internal to the package [folder]).

I should have done it in a separate commit explaining this reasoning to avoid confusion.